src/common/datatypes/Date.cpp

```diff
@@ -92,16 +97,224 @@ Date Date::operator-(int64_t days) const {
   return Date(daysSince - days);
 }
 
+void Date::addDuration(const Duration& duration) {
```

Comment:
Better to modify this function to `friend Date operator+(const Date &date, const Duration &dur)`.
eg. The implementation of expression `Duration(..)+Date(..)` will be simpler.

Reply:
Fixed.